.github/workflows/release.yml

```diff
@@ -0,0 +1,29 @@
+name: Release
+on:
+  push:
+    branches:
+      - main
+jobs:
+  test:
+    uses: ./.github/workflows/test.yml
+  release:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - uses: ./.github/actions/setup-test-env
+      - name: Build all packages
+        shell: nix develop -c bash -eo pipefail -l {0}
+        run: yarn build
+      - name: Publish @gadgetinc/api-client-core
+        uses: JS-DevTools/npm-publish@v1
```

Comment:
I'm going to assume this one is smart enough to know whether or not it needs to release? :)